overlord/confdbstate/confdbstate.go

```diff
@@ -286,9 +288,13 @@ func GetTransactionToSet(ctx *hookstate.Context, st *state.State, view *confdb.V
 	return tx, commitTx, nil
 }
 
-var ensureNow = func(st *state.State) {
-	st.EnsureBefore(0)
-}
+var (
+	ensureNow = func(st *state.State) {
+		st.EnsureBefore(0)
+	}
+
+	transactionTimeout = 2 * time.Minute
```

Comment:
This feels like quite a lot of time, Is this guarding against potential external hook delay or something internal to snapd?

Reply:
This invokes (possibly) several hooks per custodian snap, so I thought it might be good to have a guard against any broken or very slow hooks. Especially as they might have a slightly more complicated logic when saving/loading data. Hooks already have a 10m timeout but it seemed reasonable to shorten it for a read